### ISSUE TITLE
Custom melee range switch not working

### ISSUE THREAD

**Issue by lucaspicerni** [otigon/automated-jb2a-animations#467]
Custom range switching melee animations are not working. The auto-recognized swap is fine, but when I try to determine grid distance and animations manually, AA always plays the melee animations. Here what I'm getting on the console:

![image](https://user-images.githubusercontent.com/95550238/183225683-829fa759-0b3b-4fc1-8313-85f29109254a.png)

**Comment by lucaspicerni** [comment]
I noticed another issue, too: in the metric system, each square of the grid (D&D) represents 1.5 meters, so when the grid is configured this way, AA only performs the animation change from the third square. I believe he does the calculation based on 5 units of distance from the grid instead of quantities of squares, hence the error. I thought about opening a second issue, but I believe that solving this problem also solves the metric system issue, since from there it is possible to simply configure the number of squares from which the animation change happens.

**Comment by otigon** [comment]
Need some clarification on the issue here, is it only ever playing the Melee style animation and not the Range, or is it just within a certain distance because you are using a metric system?

**Comment by lucaspicerni** [comment]
There are two problems, but I believe solving the first solves both.

Problem 1: When configuring a melee animation, with the "custom" option selected in the "switch range" menu, the AA always plays the melee animation, regardless of how far the attacker is from the target:

https://user-images.githubusercontent.com/95550238/183306799-52cd5d96-e6b0-47b1-9fe4-469276a9dfe9.mp4

![image](https://user-images.githubusercontent.com/95550238/183306716-49c5415f-e656-4cae-84bc-550f65b15bd1.png)

The item is configured with the dagger animation, but from a distance it is configured as a handaxe and displays the dagger animation anyway.

Problem 2: However, when I change the setting from "custom" to "enabled", the animation type switching happens "as it should":

https://user-images.githubusercontent.com/95550238/183307111-e1529e07-0ce3-4f2e-b3c4-e77f44c206d6.mp4

![image](https://user-images.githubusercontent.com/95550238/183306852-a353a555-5edb-4f83-8cea-c6a562307782.png)

But, as you can see, although auto range switching is working, it is incompatible with the metric system as it only occurs after 3 squares away. As I said at the beginning, it wouldn't be a problem if custom range switching was working.

**Comment by otigon** [comment]
completed with A-A version 4.0.0

**Comment by lucaspicerni** [comment]
Hello! Sorry to bother you, but this bug persists. Let me clarify the question:

From what I could tell, the AA uses the grid's distance UNITS to calculate the animation switch between melee and ranged, not the grid's SQUARES.

This way, this exchange of animations is not working in the metric system, either automatically or manually configured. In the metric system, each square on the canvas represents 1.5 meters (the equivalent of 5 feet), so the animation changes only after the 4th square of distance.

This shouldn't be a problem and could be solved by configuring this switch manually, in the AA menu, as it indicates distance SQUARES, not UNITS:

![image](https://user-images.githubusercontent.com/95550238/224583542-a5d51adb-eee9-4c30-a880-2a57b1e12450.png)

However, I tested it today and the manual change is still not working, apparently considering UNITS of distance, contrary to what the menu indicates. For your testing purposes, these are my settings for a dagger:

![image](https://user-images.githubusercontent.com/95550238/224583557-2db42625-00b4-4046-94b7-6d94750fd3ce.png)

![image](https://user-images.githubusercontent.com/95550238/224583567-e999d9f1-a469-44ee-b8ab-2c5cc8c32e30.png)
